internal/backends/postgresql/metadata/pool/transaction_test.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pool
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/FerretDB/FerretDB/internal/clientconn/conninfo"
+	"github.com/FerretDB/FerretDB/internal/util/state"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
+)
+
+func TestInTransaction(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping in -short mode")
+	}
+
+	// do not run parallel, there is something weird about context canceling and parallel test
+	ctx := conninfo.Ctx(testutil.Ctx(t), conninfo.New())
+	username, password := conninfo.Get(ctx).Auth()
+
+	sp, err := state.NewProvider("")
+	require.NoError(t, err)
+
+	u := "postgres://username:password@127.0.0.1:5432/ferretdb?pool_min_conns=1"
+
+	pp, err := New(u, testutil.Logger(t), sp)
+	require.NoError(t, err)
+
+	t.Cleanup(pp.Close)
+
+	p, err := pp.Get(username, password)
+	require.NoError(t, err)
+
+	t.Cleanup(p.Close)
+
+	_, err = p.Exec(ctx, `DROP TABLE IF EXISTS t_test; CREATE TABLE t_test(s TEXT);`)
```

Comment:
I think we should use `t.Name()` to be able to track orphan table back to test

Reply:
Good idea, applied it.